packages/css/src/plugins/at-rules/parse-at-rule.ts

```diff
@@ -0,0 +1,111 @@
+import {
+  parseSituationFour,
+  parseSituationOne,
+  parseSituationThree,
+  parseSituationTwo,
+} from './parsers';
+import type { ParsedAtRule, Situations } from './types';
+
+const comparisonOperators = /(?<operator>(?:<=?)|(?:>=?)|=)\s*/;
+const comparisonOperators2 = /(?<operator2>(?:<=?)|(?:>=?)|=)\s*/;
+// Matches min-width, min-height, max-width, max-height,
+// min-device-width, min-device-height, max-device-width, max-device-height,
+// width, height, device-width, and device-height.
+const property = /(?:(?<property>((?:min|max)-)?(?:device-)?(?:width|height))\s*)/;
```

Comment:
ok i know regex is hard to read 😓 

you can treat this as being equivalent to

```
/((((min|max)-)?(device-)?(width|height))\s*)/
```

ignore the `?:` stuff, this just makes a [non-capturing group](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Regular_expressions/Non-capturing_group)

the `(?<text>` stuff is a [named capturing group](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Regular_expressions/Named_capturing_group) which lets me "store" parts of the match as values in the `match.groups` object

https://regexr.com/ is also helpful here